src/com/sap/piper/ConfigurationMerger.groovy

```diff
@@ -12,6 +14,22 @@ class ConfigurationMerger {
         return merged
     }
 
+    @NonCPS
+    def static merge(Map configs, Map configKeys, Map defaults = [:]) {
+        Map merged = [:]
+        merged.putAll(defaults)
+        if(configs != null)
+            for(String key : configKeys.keySet()){
```

Comment:
@marcusholl: merging takes place based on the keymap handed in. This way no filtering is possible for step of config values.

Merging should be done based on the default config map, right?

Reply:
@CCFenner I guess your comment refers to another comment from me where I claimed about putting a long map-like structure inside the default config(?). That mappy structure defines the keys for filtering. I was puzzled and I thought this would be step-defaults, which should come from the file holding the defaults. I already recognized my mistake and revoked my comment.